apps/gitness/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import { createBrowserRouter, RouterProvider } from 'react-router-dom'
+import { QueryClientProvider } from '@tanstack/react-query'
+import { ThemeProvider } from '@harnessio/playground'
+import { CodeServiceAPIClient } from '@harnessio/code-service-client'
+import { queryClient } from './framework/queryClient'
+import PipelineList from './pages/pipeline-list'
+import { UI_BASE_URL } from '../app_constants.ts'
+
+export default function App() {
+  React.useEffect(() => {
+    new CodeServiceAPIClient({
+      requestInterceptor: (request: Request): Request => {
+        const apiPrefix = 'api/v1/'
+
+        let newUrl = request.url
+
+        // Check if the request URL starts with the base URL
+        if (request.url.startsWith(UI_BASE_URL)) {
+          // Insert '/api/v1/' after the base URL
+          newUrl = request.url.replace(UI_BASE_URL, `${UI_BASE_URL}${apiPrefix}`)
```

Comment:
this is not what I meant 😅 
in the OATS config for code service, we need to override the base url. let's check how to make that possible. for now, we dont need this change. the original setup was cleaner.

Reply:
Done, as per discussion.